### PR TITLE
Fix processing of spaces in Trans component.

### LINKED ISSUE #113: [Trans] Blank spaces between nodes are ignored
**Describe the bug**

The extractor ignores whitespace children of `Trans` between two other children, however sometimes it shoud be ignored (if there's a line break), sometimes it should not.

**How to reproduce**

*Babel configuration:*

```
['i18next-extract', {keySeparator: null, nsSeparator: null, useI18nextDefaultValue: ['en']}],
```

*Reproduction:*

```
<Trans>
  <strong>Hello</strong>
  <strong>World 1</strong>
</Trans>

<Trans>
  <strong>Hello</strong> <strong>World 2</strong>
</Trans>

<Trans>
  <strong>Hello</strong> <MyComponent>World 3</MyComponent>
</Trans>
```

**Expected behavior**

The extracted strings should be:

```
<strong>Hello</strong><strong>World 1</strong>
<strong>Hello</strong> <strong>World 2</strong>
<strong>Hello</strong> <2>World 3</2>
```

**What actually happens**

```
<strong>Hello</strong><strong>World 1</strong>
<strong>Hello</strong><strong>World 2</strong>
<strong>Hello</strong><1>World 3</1>
```

**Your environment**

- OS (e.g. ArchLinux): `Linux 37dfc14918c4 5.3.0-24-generic #26-Ubuntu SMP Thu Nov 14 01:33:18 UTC 2019 x86_64 GNU/Linux` (within Docker)
- Plugin version (e.g. 0.3.0): `0.4.1`
- Node version (e.g. 12.13.0): ? `11.12.0`

### PR DESCRIPTION
**Pull request by gilbsgilbs**
Previously, if a JSXText node in a Trans component consisted in only one
whitespace, the extractor would strip it and ignore it for the indices
increment. This is wrong as react-i18next does take them into account.
However, react-i18next indeed ignores JSXText nodes that consist in only
whitespaces AND contains at list one linefeed. This commit makes the
extractor reflect the behavior of react-i18next.

Fixes #113

---

@pcorpet it appears that it was **not** a regression introduced in #111. This bug existed from the very beginning and is not related to basic html nodes. Thanks for bringing it to my attention. Could you have a look at my fix?

### CODE REVIEW

File: tests/__fixtures__/testTransComponent/whitespaces.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+// A single whitespace should not be stripped
+const comp2 = <Trans i18nKey="key2">
+  <span>Hello</span> <span>World 2</span>
+</Trans>;
+
+// Multiple whitespaces should not be stripped
+const comp3 = <Trans i18nKey="key3">
+  <span>Hello</span>   <span>World 3</span>
+</Trans>;
```
Comment (pcorpet) [inline tests/__fixtures__/testTransComponent/whitespaces.js]: I find it curious that there's no ending line break in some of your files where some other files have it. I suggest you add a final line break to your file here.

**Comment by gilbsgilbs** [comment]
> I think it would be nice to share some of your test files with react-i18next, to make sure they stay in sync. I don't have a quick way of doing that, I was just raising the thought.

Yes, that also applies to i18next actually. I sometimes notice regressions either because some test was all wrong from the beginning or because [react-]i18next did a (subtle) breaking change at some point. We currently have no automatic way of finding such regression because I couldn't find a way to implement it without weakening the tests in respect of parsing. One solution I had in mind at the time was to create a custom backend for i18next that would extract the keys directly from the test files (using the JS runtime we have in Jest) so that we can compare them with the keys obtained through the plugin. Yet, this solution had the downside of preventing you to make use of any type of complex structure (such as conditions or functions) which you do want to test in a parser.

Therefore it would be possible to have such test, but might require way more work that I'm currently able to provide in my spare time. I think the more the plugin will be used, the more we'll tend to a *status quo* situation where we know (from the community) that the tests (and as the consequence the plugin) do reflect the reality. Not a perfect solution, but probably good enough (unless somebody wants to improve :angel:). That's why bug reports such as #113 matters :+1: .

File: tests/__fixtures__/testTransComponent/whitespaces.js (hunk 1 above)
Comment (gilbsgilbs) [reply tests/__fixtures__/testTransComponent/whitespaces.js]: True. I'll add them to all files in another branch then.

**Comment by gilbsgilbs** [comment]
Thanks for the prompt review btw. Just released this on NPM (v0.4.2).

**Pull request merged**